### DISCUSSION TITLE
Random forest model persistence

### DISCUSSION THREAD

**Discussion by jblindsay** [smartcorelib/smartcore#264] [Q&A]
I am trying to use model persistence with a random forest regression model, however, I am having troubles with the decoding step. I can't understand what the generic parameters of my RF model, TX, TY, X, Y, should be. The example of model persistence in the Smartcore user manual is for a kNN classifier. Would it be possible for anyone to provide a snippet example of a RF regression model with persistence so that I can try to figure this out in my case? Thanks.

**Comment by Mec-iS** [comment]
Hi!
Thanks for using `smartcore`.

Please provide a link to the model persistence example you mentioned.

Have you tried to take a look to the tests module in the regressor's module, you can find them [here](https://github.com/smartcorelib/smartcore/blob/development/src/ensemble/random_forest_regressor.rs#L526). If your question is not addressed in the tests, let me know and I will provide to add it. You can also try to add it yourself, please consider opening a PR if you manage.

**Comment by jblindsay** [comment]
Hi,
Thanks for creating `smartcore`! The example that I've seen in the user manual is [here](https://smartcorelib.org/user_guide/model_selection.html#model-persistence). I'll take a look at the tests module as suggested. Thanks again.

**Comment by jblindsay** [comment]
Just to let you know, the example in the test worked perfectly to help me resolve my issue. Much appreciated once again!
